From 741a999088dc227b5a83e7e09bcf27b51a6c65db Mon Sep 17 00:00:00 2001
From: Daniel Mueller <deso@posteo.net>
Date: Sun, 18 Nov 2018 18:41:16 -0800
Subject: [PATCH] Fall back to using racer for definition lookup by default

The rls seems to have problems looking up the definitions of symbols in
the standard library. For racer we have installed the std lib sources
and set RUST_SRC_PATH to point to them.
With this change we adjust the rls to fall back to using racer for
definition lookups unless explicitly changed.
---
 src/tools/rls/src/config.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tools/rls/src/config.rs b/src/tools/rls/src/config.rs
index d171ad..198eb4 100644
--- a/src/tools/rls/src/config.rs
+++ b/src/tools/rls/src/config.rs
@@ -177,7 +177,7 @@ impl Default for Config {
             unstable_features: false,
             wait_to_build: None,
             show_warnings: true,
-            goto_def_racer_fallback: false,
+            goto_def_racer_fallback: true,
             clear_env_rust_log: true,
             build_on_save: false,
             use_crate_blacklist: true,
-- 
2.19.2

